src/lib/deployWorkers.ts

```diff
@@ -765,8 +773,12 @@ async function resolveWorker({
     )}`,
   );
 
-  const services: Upload_deployArgConfig["workers"][number]["config"]["services"] =
-    (workerConfig.services ?? []).map((serviceName) => {
+  const servicesWithUnresolvedMemoryLimit = (workerConfig.services ?? []).map(
+    (
+      serviceName,
+    ): Upload_deployArgConfig["workers"][number]["config"]["services"][number] & {
+      total_memory_limit: string | undefined;
+    } => {
```

Comment:
Let's have a separate type for this. Probably even in this file above it there is no need to reuse it